src/ng/directive/ngStyle.js

```diff
@@ -54,7 +54,11 @@
 var ngStyleDirective = ngDirective(function(scope, element, attr) {
   scope.$watchCollection(attr.ngStyle, function ngStyleWatchAction(newStyles, oldStyles) {
     if (oldStyles && (newStyles !== oldStyles)) {
-      forEach(oldStyles, function(val, style) { element.css(style, '');});
+      forEach(oldStyles, function(val, style) {
+          if (!(newStyles && newStyles.hasOwnProperty(style))) {
```

Comment:
What if `newStyles` does have the property, but the value is `undefined` or `null`?

Reply:
Right, in that case the style should be cleared. Fixed in 5d74f0f2884f0455a6735f58966698ea70f609c9.